README.md

```diff
@@ -1,20 +1,236 @@
+  destination: 'FS.event'
+ }
+];
+```
+
+The above sample contains two rules that respectively send all properties in the `digitalData.user.profile` object to the function `FS.setUserVars` and sends specific properties from the `digitalData.transaction` object to the `FS.event` function.  The mechanics of how this is done is explained in the [Operator Tutorial](https://github.com/fullstorydev/fullstory-data-layer-observer/tree/main/docs/operator_tutorial.md) in greater detail.
+
+> **Tip**: While not required, a best practice is to include an `id` in every rule.  The `id` should uniquely identify the rule for troubleshooting purposes.  A `description` is also optional but often helps explain the intent of a rule.
+
+## Rule Options
+
+Each rule provides a set of options for configuration.  Options with an asterisk are required.
+
+| Option | Default | Description |
+| ------ | ------- | ----------- |
+| `source` * | `undefined` | Data layer source object using selector syntax. |
+| `destination` * | `undefined` | Destination function using selector syntax. |
+| `debug` | `false` | Set to true if the rule should print debug for each operator transformation. |
+| `description` | `undefined` | Text description of the rule. |
+| `id` | `undefined` | Unique identifier for the rule. |
+| `operators` | `[]` | List of operators that transform data before a destination. |
+| `readOnLoad` | `false` | Rule-specific override for `window[‘_dlo_readOnLoad’]`. |
+| `url` | `undefined` | Specifies a regular expression that enables the rule when the page URL matches. |
+
+> **Tip:** Use `url` to limit when data is read from a data layer and enhance performance.
+
+## Data Handling
+
+In version 1.0.x, data from a data layer is evaluated when DLO loads on the page.  The configuration option `window['_dlo_readOnLoad'] = true;` should be used for static data layers, which should exist on the page prior to loading DLO.  In the very near future, monitoring of dynamic changes in a data layer will be introduced.
+
+## Preview and Debugging Rules
+
+Rule writing can sometimes take a few attempts to get it right.  Fortunately, the following two options can help.
+
+- When the configuration option `_dlo_previewMode` is set to `true`, output will be written to `console.log` rather than the `destination`.
+- When a particular rule's `debug` property is set to `true`, the incremental transformations performed by `operators` and additional logging will be written to `console.debug`.
+
+Viewing the JavaScript console with the above option set to `true` prints the following example.
+
+```javascript
+dataLayer.transaction[!(items)] handleData entry
+[{'transactionID':'2325112','total':{'grandTotal':'45.59','subTotal':'37.02','shippingTotal':'5.99','totalTax':'2.58','promotionCodes':[],'selectedPayment':'VISA'}}]
+dlo-debug.js:1136   [0] flatten output (numKeys=7 sizeOfValues=94 sizeOfPayload=342)
```

Comment:
`numKeys=7 sizeOfValues=94 sizeOfPayload=342`

Can you also describe what these fields mean?